File: AutomationTesting/python/pages/checkout_page.py
from selenium.webdriver.common.by import By

class CheckoutPage:
    def __init__(self, driver):
        self.driver = driver
        self.cart_button = (By.XPATH, "//a[@href='/view_cart']")
        self.proceed_to_checkout_button = (By.XPATH, "//a[contains(text(),'Proceed To Checkout')]")
        self.cart_items = (By.CLASS_NAME, "cart_description")

    def go_to_cart(self):
        self.driver.find_element(*self.cart_button).click()

    def click_proceed_to_checkout(self):
        self.driver.find_element(*self.proceed_to_checkout_button).click()

    def get_cart_items(self):
        return self.driver.find_elements(*self.cart_items)

File: AutomationTesting/Java/src/main/java/pages/checkout/CheckoutPage.java
package pages.checkout;

import pages.base.BasePage;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import pages.payment.PaymentPage;

public class CheckoutPage extends BasePage {

    @FindBy(xpath = "//a[@href='/payment' and @class='btn btn-default check_out']")
    private WebElement placeOrderButton;

    public CheckoutPage(WebDriver driver) {
        super(driver);
    }

    public PaymentPage placeOrder() {
        click(placeOrderButton);
        return new PaymentPage(driver);
    }

    public boolean isProductInCheckout(int productId) {
        try {
            WebElement productElement = driver.findElement(
                    By.xpath("//tr[@id='product-" + productId + "']//a[@href='/product_details/" + productId + "']"));
            return productElement.isDisplayed();
        } catch (Exception e) {
            return false;
        }
    }

    public int getProductQuantity(int productId) {
        WebElement quantityElement = driver.findElement(
                By.xpath("//tr[@id='product-" + productId + "']//td[@class='cart_quantity']/button"));
        return Integer.parseInt(getText(quantityElement));
    }

    public String getProductName(int productId) {
        WebElement nameElement = driver.findElement(
                By.xpath("//tr[@id='product-" + productId + "']//td[@class='cart_description']/h4/a"));
        return getText(nameElement);
    }
}